### PR TITLE
bpo-37471: mmap module adding FreeBSD specific flag into the constants

### PR DESCRIPTION
**Pull request by ghost**
https://bugs.python.org/issue37471

### CODE REVIEW

**Review by vstinner — COMMENTED** [review]
Would it be possible to document the constant in https://docs.python.org/dev/library/mmap.html ?

It seems like other MAP_xxx constants are not documented yet and so should be documented as well.

**Comment by dcarlier-afilias** [comment]
They are (shortly) in the flags section. Ok for adding the new constant too.

**Comment by vstinner** [comment]
The minimum doc should be a list of contants, like: https://docs.python.org/dev/library/os.html#os.O_RDONLY

**Comment by dcarlier-afilias** [comment]
Fair enough ! ought to be before madvise's I think.

File: Doc/library/mmap.rst [hunk 1]
```diff
@@ -81,7 +81,11 @@ To map anonymous memory, -1 should be passed as the fileno along with the length
    private copy-on-write mapping, so changes to the contents of the mmap
    object will be private to this process, and :const:`MAP_SHARED` creates a
    mapping that's shared with all other processes mapping the same areas of
-   the file.  The default value is :const:`MAP_SHARED`.
+   the file. :const:`MAP_ALIGNED_SUPER` (FreeBSD only) allows to map to
+   pages of 2MB (x86) or so, it is enabled by default, at boot time,
```
Comment (vstinner) [inline Doc/library/mmap.rst]:
```suggestion
   pages of 2 MiB (x86) or so, it is enabled by default, at boot time,
```

Is it x86 (32-bit) or x86-64 (64-bit)?

File: Misc/NEWS.d/next/Library/2019-07-01-14-27-17.bpo-37471._EFODL.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Adding new 'MAP_ALIGNED_SUPER' FreeBSD constant.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2019-07-01-14-27-17.bpo-37471._EFODL.rst]:
```suggestion
Adding new :data:`mmap.MAP_ALIGNED_SUPER` FreeBSD constant.
```

File: Doc/library/mmap.rst [hunk 3]
```diff
@@ -307,6 +311,19 @@ To map anonymous memory, -1 should be passed as the fileno along with the length
...
+MAP_* Constants
++++++++++++++++
+
+.. data:: MAP_ANONYMOUS
+          MAP_ANON
+          MAP_SHARED
+          MAP_PRIVATE
+          MAP_DENYWRITE
+          MAP_EXECUTABLE
+          MAP_ALIGNED_SUPER
```
Comment (vstinner) [inline Doc/library/mmap.rst:325]: You have to document that MAP_ALIGNED_SUPER is new, example:

```
.. versionadded:: 3.9
   Add :data:`MAP_ALIGNED_SUPER` constant.
```

File: Doc/library/mmap.rst (hunk 1 above)
Comment (ghost) [reply Doc/library/mmap.rst]: both.

Comment (vstinner) [reply Doc/library/mmap.rst]: Write maybe (x86, x86-64) in this case... But I'm not sure if Python documentation should go too far in the description, since FreeBSD may decide to add support for MAP_ALIGNED_SUPER for other platforms.

File: Doc/library/mmap.rst [hunk 4]
```diff
@@ -81,7 +81,11 @@ To map anonymous memory, -1 should be passed as the fileno along with the length
    private copy-on-write mapping, so changes to the contents of the mmap
    object will be private to this process, and :const:`MAP_SHARED` creates a
    mapping that's shared with all other processes mapping the same areas of
-   the file.  The default value is :const:`MAP_SHARED`.
+   the file. :const:`MAP_ALIGNED_SUPER` (FreeBSD only) allows to map to
+   pages of 2MB (x86) or so, it is enabled by default, at boot time,
+   in most of architectures and exists since 2008.
+   :ref:`MAP_* constants <mmap-constants>`.
```
Comment (vstinner) [inline Doc/library/mmap.rst]: "MAP_* constants" is not really a sentence. Maybe move it after the default, and write something like:

```
The default value is :const:`MAP_SHARED`. 
See :ref:`MAP_* constants <mmap-constants>`.
```

File: Doc/library/mmap.rst (hunk 1 above)
Comment (ghost) [reply Doc/library/mmap.rst]: It supports more than x86 already in fact but you re right I ll try to find more concise wording

File: Doc/library/mmap.rst [hunk 5]
```diff
@@ -341,3 +359,6 @@ MADV_* Constants
    Availability: Systems with the madvise() system call.
 
    .. versionadded:: 3.8
+
+   .. versionadded:: 3.9
+      Add :data:`MAP_ALIGNED_SUPER` constant.
```
Comment (vstinner) [inline Doc/library/mmap.rst]: Can you move this to the your new  MAP_* Constants  section?

File: Misc/NEWS.d/next/Library/2019-07-05-20-54-46.bpo-35711.xYuDRM.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Adding new :data:`mmap.MAP_CONCEAL` OpenBSD constant.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2019-07-05-20-54-46.bpo-35711.xYuDRM.rst]: Can you try to merge the 2 changelog entries? Put content of the 2nd entry in the first one, remove the second file.

File: Doc/library/mmap.rst [hunk 7]
```diff
@@ -307,6 +311,20 @@ To map anonymous memory, -1 should be passed as the fileno along with the length
...
++++++++++++++++
+
+.. data:: MAP_ANONYMOUS
+          MAP_ANON
+          MAP_SHARED
+          MAP_PRIVATE
+          MAP_DENYWRITE
+          MAP_EXECUTABLE
+          MAP_ALIGNED_SUPER
+          MAP_CONCEAL
```
Comment (vstinner) [inline Doc/library/mmap.rst:326]: You have to document that MAP_CONCEAL is new.

File: Doc/library/mmap.rst [hunk 8]
```diff
@@ -81,7 +81,11 @@ To map anonymous memory, -1 should be passed as the fileno along with the length
    private copy-on-write mapping, so changes to the contents of the mmap
    object will be private to this process, and :const:`MAP_SHARED` creates a
    mapping that's shared with all other processes mapping the same areas of
-   the file.  The default value is :const:`MAP_SHARED`.
+   the file. :const:`MAP_ALIGNED_SUPER` (FreeBSD only) to use
+   super pages feature, enabled by default in most of architectures
+   and exists since 2008.
```
Comment (vstinner) [inline Doc/library/mmap.rst]: I don't understand this sentence :-/ I suggest:

```
:const:`MAP_ALIGNED_SUPER` (FreeBSD only) uses super pages feature,
it is enabled by default in most of architectures and exists since 2008.
```
